File: rb/features.rb
# Genrenator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module GenrenatorFeatures
  def self.make_feature(name)
    case name
    when "base"
      GenrenatorBaseFeature.new
    when "ratelimit"
      GenrenatorRatelimitFeature.new
    when "retry"
      GenrenatorRetryFeature.new
    when "test"
      GenrenatorTestFeature.new
    when "timeout"
      GenrenatorTimeoutFeature.new
    else
      GenrenatorBaseFeature.new
    end
  end
end

File: py/features.py
# Genrenator SDK feature factory

from feature.base_feature import GenrenatorBaseFeature
from feature.test_feature import GenrenatorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GenrenatorBaseFeature(),
        "test": lambda: GenrenatorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
